--- a/support-files/CMakeLists.txt	2019-06-21 19:47:15.000000000 +0200
+++ b/support-files/CMakeLists.txt	2019-06-21 19:51:30.759640655 +0200
@@ -155,6 +155,23 @@
       ENDIF()

     ENDIF()
+  ELSE()
+    IF(INSTALL_SYSTEMD_SYSUSERSDIR)
+      CONFIGURE_FILE(sysusers.conf.in
+              ${CMAKE_CURRENT_BINARY_DIR}/sysusers.conf @ONLY)
+      INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/sysusers.conf
+              DESTINATION ${INSTALL_SYSTEMD_SYSUSERSDIR}
+              RENAME mariadb.conf COMPONENT Server)
+    ENDIF()
+
+    IF(INSTALL_SYSTEMD_TMPFILESDIR)
+      get_filename_component(MYSQL_UNIX_DIR ${MYSQL_UNIX_ADDR} DIRECTORY)
+      CONFIGURE_FILE(tmpfiles.conf.in
+              ${CMAKE_CURRENT_BINARY_DIR}/tmpfiles.conf @ONLY)
+      INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/tmpfiles.conf
+              DESTINATION ${INSTALL_SYSTEMD_TMPFILESDIR}
+              RENAME mariadb.conf COMPONENT Server)
+    ENDIF()
   ENDIF()

   IF (INSTALL_SYSCONFDIR)
