From a8e1cf1899cefad0f77196c161b615c2975418e4 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 15 Jan 2019 21:22:56 +0100
Subject: [PATCH] Makefile.in: Move D-Bus system.d into datadir

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 2d1d07f..de99daf 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -21,7 +21,7 @@ AM_LDFLAGS = $(GCLDFLAGS)
 ACLOCAL_AMFLAGS = -I m4
 
 policykitdir = $(datadir)/polkit-1/actions/
-dbussystemdir = $(sysconfdir)/dbus-1/system.d/
+dbussystemdir = $(datadir)/dbus-1/system.d/
 dbusservicedir = $(datadir)/dbus-1/system-services/
 
 EXTRA_DIST = \
