diff -Naurp ModemManager-1.8.2-orig/configure.ac ModemManager-1.8.2/configure.ac
--- ModemManager-1.8.2-orig/configure.ac	2018-09-22 04:01:11.000000000 -0500
+++ ModemManager-1.8.2/configure.ac	2018-12-13 22:30:59.307184516 -0600
@@ -237,6 +237,34 @@ esac

 AM_CONDITIONAL(WITH_SYSTEMD_SUSPEND_RESUME, test "x$with_systemd_suspend_resume" = "xyes")

+PKG_CHECK_MODULES(LIBELOGIND, [libelogind >= 209],[have_libelogind=yes],[have_libelogind=no])
+AC_ARG_WITH(elogind-suspend-resume,
+            AS_HELP_STRING([--with-elogind-suspend-resume=no|yes],
+                           [Enable elogind suspend/resume support [[default=auto]]]),,
+            [with_elogind_suspend_resume=auto])
+
+if test "x$with_elogind_suspend_resume" = "xauto"; then
+    if test "x$have_libelogindd" = "xyes"; then
+        with_elogind_suspend_resume=yes
+    else
+        with_elogind_suspend_resume=no
+    fi
+fi
+
+case $with_elogind_suspend_resume in
+    yes)
+        if test "x$have_libelogind" = "xno"; then
+                    AC_MSG_ERROR(libelogind development headers are required)
+            fi
+        AC_DEFINE(WITH_ELOGIND_SUSPEND_RESUME, 1, [Define if you have elogind suspend-resume support])
+        ;;
+    *)
+        with_elogind_suspend_resume=no
+        ;;
+esac
+
+AM_CONDITIONAL(WITH_ELOGIND_SUSPEND_RESUME, test "x$with_elogind_suspend_resume" = "xyes")
+
 dnl-----------------------------------------------------------------------------
 dnl systemd journal support
 dnl
@@ -448,6 +476,7 @@ echo "
       mbim:                    ${with_mbim}
       qmi:                     ${with_qmi}
       systemd suspend/resume:  ${with_systemd_suspend_resume}
+      elogind suspend/resume:  ${with_elogind_suspend_resume}
       systemd journal:         ${with_systemd_journal}

     Miscellaneous:
diff -Naurp ModemManager-1.8.2-orig/src/Makefile.am ModemManager-1.8.2/src/Makefile.am
--- ModemManager-1.8.2-orig/src/Makefile.am	2018-09-22 02:25:43.000000000 -0500
+++ ModemManager-1.8.2/src/Makefile.am	2018-12-13 22:32:24.251757477 -0600
@@ -363,6 +363,12 @@ if WITH_SYSTEMD_SUSPEND_RESUME
 ModemManager_SOURCES += mm-sleep-monitor.h mm-sleep-monitor.c
 endif

+# Additional suspend/resume support via elogind
+if WITH_ELOGIND_SUSPEND_RESUME
+ModemManager_SOURCES += mm-sleep-monitor.h mm-sleep-monitor.c
+endif
+
+
 # Additional QMI support in ModemManager
 if WITH_QMI
 ModemManager_SOURCES += \
